application/frontend/controllers/ResetController.php

```diff
@@ -337,8 +336,8 @@ public function actionValidate($uid)
      */
     protected function getCodeFromRequestBody(): string
     {
-        $code = \Yii::$app->request->getBodyParam('code', null);
-        if ($code === null) {
+        $code = trim(\Yii::$app->request->getBodyParam('code', null));
```

Comment:
same null trim danger. -- but this default to empty string instead of null could be a easy cleanup.